src/main/java/com/gradle/enterprise/conventions/customvalueprovider/LocalBuildCustomValueProvider.java

```diff
@@ -21,16 +22,30 @@ public boolean isEnabled() {
     @Override
     public void accept(Settings settings, BuildScanExtension buildScan) {
         buildScan.tag("LOCAL");
-        if (Stream.of("idea.registered", "idea.active", "idea.paths.selector").anyMatch(it -> getConventions().systemPropertyProvider(it).isPresent())) {
+        if (isRunningInIdea()) {
             buildScan.tag("IDEA");
             String ideaVersion = getConventions().systemPropertyProvider("idea.paths.selector").getOrNull();
             if (ideaVersion != null) {
                 buildScan.value(IDEA_VERSION, ideaVersion);
             }
         }
+        captureRevisionInBackground(getConventions(), settings, buildScan);
+    }
+
+    private boolean isRunningInIdea() {
+        return Stream.of("idea.registered", "idea.active", "idea.paths.selector")
+            .anyMatch(it -> getConventions().systemPropertyProvider(it)
+                .isPresent());
+    }
 
-        GradleEnterpriseConventions.execAndGetStdout(settings.getRootDir(), "git", "rev-parse", "HEAD")
-            .ifPresent(commitId -> getConventions().setCommitId(settings.getRootDir(), buildScan, commitId));
+    private static void captureRevisionInBackground(
+        GradleEnterpriseConventions conventions, Settings settings, BuildScanExtension buildScan
+    ) {
+        File rootDir = settings.getRootDir(); // Do not capture the whole settings instance or configuration caching may
+        // fail.
+        buildScan.background(__ ->
```

Comment:
```suggestion
        buildScan.background(buildScan ->
```

This needs to use the build scan from the closure, if it doesn't, then it will not update.

Reply:
>This needs to use the build scan from the closure, if it doesn't, then it will not update on a CC hit.

Done, thanks. I've updated both callsites